src/ConfigManager.cpp

```diff
@@ -0,0 +1,99 @@
+#include "ConfigManager.hpp"
+#include <cstdlib>
+#include <hyprlang.hpp>
+#include <hyprutils/path/Path.hpp>
+#include <string>
+#include <sys/ucontext.h>
+#include "helpers/Log.hpp"
+
+static std::string getMainConfigPath() {
+    static const auto paths = Hyprutils::Path::findConfig("hyprsunset");
+    if (paths.first.has_value())
+        return paths.first.value();
+    else
+        throw std::runtime_error("Could not find config in HOME, XDG_CONFIG_HOME, XDG_CONFIG_DIRS or /etc/hypr.");
```

Comment:
please allow for running without a config, this is unnesessary. Hyprlang can even take a missing config without errors, see the parameters given to the main manager class